RecoMuon/L2MuonSeedGenerator/plugins/L2MuonSeedGeneratorFromL1TkMu.cc

```diff
@@ -388,21 +454,21 @@ const TrajectorySeed *L2MuonSeedGeneratorFromL1TkMu::associateOfflineSeedToL1(
                          << ", phi=" << offseedTsos.globalMomentum().phi()
                          << ", eta=" << offseedTsos.globalMomentum().eta() << ")" << std::endl
                          << std::endl;
-      double newDr = deltaR(newTsos.globalPosition().eta(),
-                            newTsos.globalPosition().phi(),
-                            offseedTsos.globalPosition().eta(),
-                            offseedTsos.globalPosition().phi());
-      LogDebug(metlabel) << "   -- DR = " << newDr << std::endl;
-      if (newDr < dRcone && newDr < bestDr) {
-        LogDebug(metlabel) << "          --> OK! " << newDr << std::endl << std::endl;
+      double newDr2 = deltaR2(newTsos.globalPosition().eta(),
+                              newTsos.globalPosition().phi(),
+                              offseedTsos.globalPosition().eta(),
+                              offseedTsos.globalPosition().phi());
+      LogDebug(metlabel) << "   -- DR = " << newDr2 << std::endl;
+      if (newDr2 < dRcone * dRcone && newDr2 < bestDr2) {
```

Comment:
```suggestion
      if (newDr2 < bestDr2 && newDr2 < dRcone * dRcone) {
```
can avoid computing the product after the first iteration.

By the way, is it granted that `dRcone` is always >=0 here?

Reply:
I made it return `nullptr` if `dRcone < 0.`